dev/DynamicDependency/appmodel_msixdynamicdependency.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <appmodel.h>
+
+// Modern variants of appmodel.h
+namespace appmodel
+{
+inline std::wstring GetPackagePath(PCWSTR packageFullName)
+{
+    // Paths can be long but typically short(ish). We can use a quick fixed buffer
+    // as an optimization and fallback to dynamic allocation if need be
+    WCHAR path[MAX_PATH];
+    UINT32 pathLength{ ARRAYSIZE(path) };
+    const auto rc{ ::GetPackagePathByFullName(packageFullName, &pathLength, path) };
```

Comment:
wil has a details helper that does this two-phase-with-allocation and can produce wstrings.